### ISSUE TITLE
[0.9.5.1] drop retention does not delete shards

### ISSUE THREAD

**Issue by exeral** [influxdata/influxdb#5394]
Hello,

shortly: I wiped the data by dropping default RP and replacing by a new one
problem: the shards corresponding to the deleted RP are not auto deleted and still consume disk space.

old situation:
```
name	duration	replicaN	default
default	720h0m0s	1		true
2day	48h0m0s		1		false
1week	168h0m0s	1		false
1month	720h0m0s	1		false
1year	8736h0m0s	1		false
```

new default RP:
```
> ALTER RETENTION POLICY "2day" ON mydb DEFAULT
> show retention policies on mydb
name	duration	replicaN	default
default	720h0m0s	1		false
2day	48h0m0s		1		true
1week	168h0m0s	1		false
1month	720h0m0s	1		false
1year	8736h0m0s	1		false
```

drop the old RP:
```
> DROP RETENTION POLICY "default" ON mydb
> show retention policies on mydb
name	duration	replicaN	default
2day	48h0m0s		1		true
1week	168h0m0s	1		false
1month	720h0m0s	1		false
1year	8736h0m0s	1		false
```

```
[retention] 2016/01/19 16:01:41 retention policy enforcement check commencing
[retention] 2016/01/19 16:01:41 retention policy shard deletion check commencing
[shard-precreation] 2016/01/19 16:07:04 Precreation service terminating
[shard-precreation] 2016/01/19 16:08:03 Starting precreation service with check interval of 10m0s, advance period of 30m0s
[monitor] 2016/01/19 16:08:03 Storing statistics in database '_internal' retention policy '', at interval 10s
[retention] 2016/01/19 16:08:03 Starting retention policy enforcement service with check interval of 30m0s
[retention] 2016/01/19 16:38:03 retention policy enforcement check commencing
[retention] 2016/01/19 16:38:03 retention policy shard deletion check commencing
```

the retention check has been triggered but nothing hapenned
I restarted influxdb
the retention check has been triggered again but still nothing deleted

```
> show shards
name: mydb
------------------
id	database	retention_policy	shard_group	start_time		end_time	expiry_time		owners
177	mydb	2day			177		2016-01-19T00:00:00Z	2016-01-20T00:00:00Z	2016-01-22T00:00:00Z	1
178	mydb	1week			178		2016-01-19T00:00:00Z	2016-01-20T00:00:00Z	2016-01-27T00:00:00Z	1
179	mydb	1month			179		2016-01-19T00:00:00Z	2016-01-20T00:00:00Z	2016-02-19T00:00:00Z	1
180	mydb	1year			180		2016-01-18T00:00:00Z	2016-01-25T00:00:00Z	2017-01-23T00:00:00Z	1
```
```
# ls -alhR /var/lib/influxdb/data/mydb/
/var/lib/influxdb/data/mydb/:
total 32K
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 .
drwxr-xr-x 5 influxdb influxdb 4.0K 2015-10-12 14:50 ..
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:37 1month
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:37 1week
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:38 1year
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:35 2day
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 00:00 default
drwx------ 2 influxdb influxdb 4.0K 2016-01-07 15:28 one_month

/var/lib/influxdb/data/mydb/1month:
total 312K
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:37 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb 512K 2016-01-19 16:44 179

/var/lib/influxdb/data/mydb/1week:
total 572K
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:37 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb 1.0M 2016-01-19 16:48 178

/var/lib/influxdb/data/mydb/1year:
total 308K
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:38 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb 512K 2016-01-19 16:18 180

/var/lib/influxdb/data/mydb/2day:
total 8.7M
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 15:35 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb  16M 2016-01-19 16:49 177

/var/lib/influxdb/data/mydb/default:
total 1.9G
drwx------ 2 influxdb influxdb 4.0K 2016-01-19 00:00 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb 512M 2016-01-19 16:08 107
-rw-r--r-- 1 influxdb influxdb 512M 2016-01-19 16:08 116
-rw-r--r-- 1 influxdb influxdb 256M 2016-01-19 16:08 125
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 140
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 144
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 148
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 152
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 156
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 160
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 164
-rw-r--r-- 1 influxdb influxdb  64M 2016-01-19 16:08 168
-rw-r--r-- 1 influxdb influxdb  32M 2016-01-19 16:08 172
-rw-r--r-- 1 influxdb influxdb 512M 2016-01-19 16:08 98

/var/lib/influxdb/data/mydb/one_month:
total 4.0M
drwx------ 2 influxdb influxdb 4.0K 2016-01-07 15:28 .
drwx------ 8 influxdb influxdb 4.0K 2016-01-19 15:38 ..
-rw-r--r-- 1 influxdb influxdb 4.0M 2016-01-19 16:08 131
```

InfluxDB 0.9.5.1
bz1 engine

**Comment by exeral** [comment]
Can I safely delete the files on filesystem corresponding to the deleted retention policy ?

**Comment by jwilder** [comment]
I don't think this is storage engine specific.  The drop retention policy is probably only deleting the RP from the meta store and not on disk.

**Comment by jwilder** [comment]
@exeral Yes.

**Comment by e-dard** [comment]
Fixed via #5691.
